File: src/main/java/BTMaker/BTMaker/SpriteType.java
package BTMaker.BTMaker;

public enum SpriteType {
    STATIC(0),
    COMPOUND(1),
    ANIMATED(2);

    public final byte code;

    SpriteType(int code) {
        this.code = (byte) code;
    }

    public static SpriteType fromCode(byte code) {
        for (SpriteType type : values()) {
            if (type.code == code) {
                return type;
            }
        }
        throw new IllegalArgumentException("Unknown SpriteType code: " + code);
    }
}

File: src/main/java/BTMaker/BTMaker/resources/SpriteType.kt
package BTMaker.BTMaker.resources

enum class SpriteType {
    SIMPLE,
    COMPOUND,
    ANIMATED;

    companion object {
        fun fromCode(code: Number): SpriteType {
            return when (code) {
                0 -> SIMPLE
                1 -> COMPOUND
                2 -> ANIMATED
                else -> throw IllegalArgumentException("Unknown resource type code: $code")
            }
        }
    }
}